### PR TITLE
feat: parse path for group name

### PR DESCRIPTION
**Pull request by GraceRuan**

### CODE REVIEW

File: config/templates/hashicorp_vault/lua/audit.lua [hunk 1]
```diff
@@ -245,6 +252,29 @@ function path_to_service_target(tag, timestamp, record)
...
         end
+        if string.sub(path, 0, 7) == "groups/" then
+            path_segment = parse_string(path,"[^/]+")
+            local groupname = path_segment[3]
+            local service = path_segment[4]            
+            if groupname ~= nil then
+                record["source.user.group.name"] = groupname
+                code = 2
+            end
+            if service ~= nil then
```
Comment (mbystedt) [inline config/templates/hashicorp_vault/lua/audit.lua:263]: I don't think there's any convention (or enforcement) that the path segments under groups have any particular meaning.

File: config/templates/hashicorp_vault/lua/audit.lua [hunk 2]
```diff
@@ -219,16 +219,23 @@ pathEnvToStandardEnv["prod"] = "production"
 pathEnvToStandardEnv["test"] = "test"
 pathEnvToStandardEnv["dev"] = "development"
 
+function parse_string(str, pattern)
+    local path_arr={}
```
Comment (mbystedt) [inline config/templates/hashicorp_vault/lua/audit.lua:223]: Make sure this PR follows the code style.

File: config/templates/hashicorp_vault/lua/audit.lua (hunk 1 above)
Comment (GraceRuan) [reply config/templates/hashicorp_vault/lua/audit.lua:253]: Remove group name

File: config/templates/hashicorp_vault/lua/audit.lua (hunk 2 above)
Comment (GraceRuan) [reply config/templates/hashicorp_vault/lua/audit.lua]: Do not need use separate function to do parsing

**Comment by GraceRuan** [comment]
This group path parse will be followed up after group schema being enabled in vault.

**Pull request closed without merging**